src/datasources.js

```diff
@@ -225,11 +221,59 @@ define([
         return widget;
     }
 
+    function fixtureWidget(mug, options, labelText) {
```

Comment:
I think this widget will eventually be used for data sources other than fixtures as well. Would it make sense to name it something like `dataSourceWidget` and rename the existing widget with that name to `advancedDataSourceWidget`?